scopesim/optics/image_plane_utils.py

```diff
@@ -525,28 +528,15 @@ def rescale_imagehdu(imagehdu: fits.ImageHDU, pixel_scale: float | u.Quantity,
             logger.warning("imagehdu.data.ndim is %d, but wcs.naxis with key "
                            "%s is %d, both should be equal.",
                            imagehdu.data.ndim, ww.wcs.alt, ww.naxis)
-            # TODO: could this be ww = ww.sub(2) instead? or .celestial?
-            # ww = WCS(imagehdu.header, key=key, naxis=imagehdu.data.ndim)
 
         if any(ctype != "LINEAR" for ctype in ww.wcs.ctype):
             logger.warning("Non-linear WCS rescaled using linear procedure.")
 
-        new_crpix = (zoom + 1) / 2 + (ww.wcs.crpix - 1) * zoom
-        #ew_crpix = np.round(new_crpix * 2) / 2  # round to nearest half-pixel
-        logger.debug("new crpix %s", new_crpix)
-        ww.wcs.crpix = new_crpix
+        ww.wcs.crpix[:2] = (zoom[:2] + 1) / 2 + (ww.wcs.crpix[:2] - 1) * zoom[:2]
```

Comment:
I had some trouble understanding where this equation came from. Maybe we can add the derivation as a comment to the code?
```suggestion
        # In the old coordinate system the following holds (assuming linearity):
        #   VAL = CRVAL + (PIX - CRPIX) * CDELT
        # Denoting the zoomed coordinate system with primes we get:
        #   VAL' = CRVAL' + (PIX' - CRPIX') * CDELT'
        # Where by definition CDELT' = CDELT / ZOOM, and CRVAL' = CRVAL.
        #
        # There is always a fixed point in such a transformation.
        # All values refer to the center of the pixels, so it is the 
        # edge of the first pixel whose VAL is conserved.
        # That is, for PIX = 1/2, PIX' = 1/2, and VAL' = VAL.
        # 
        # Filling the above two equations with the values for the
        # edge of the first pixel yields:
        #   VAL = CRVAL + (1/2 - CRPIX) * CDELT
        #   VAL = CRVAL + (1/2 - CRPIX') * CDELT / ZOOM
        # Equating these equations allows us to solve for CRPIX':
        #   (1/2 - CRPIX) * CDELT = (1/2 - CRPIX') * CDELT / ZOOM
        #   (1/2 - CRPIX) * ZOOM = 1/2 - CRPIX'
        #   CRPIX' = 1/2 - (1/2 - CRPIX) * ZOOM
        #   CRPIX' = 1/2 + (CRPIX - 1/2) * ZOOM
        #   CRPIX' = 1/2 + CRPIX * ZOOM - ZOOM / 2
        # Which is (effectively) the same as the equation below.
        ww.wcs.crpix[:2] = (zoom[:2] + 1) / 2 + (ww.wcs.crpix[:2] - 1) * zoom[:2]
```

Reply:
Thanks! I found my derivation a bit too verbose too.

I was wondering whether there is a way to just let astropy.wcs do this for us. But that isn't really relevant now we already have this nicely working.